### ISSUE TITLE
Implement centralized training

### ISSUE THREAD

**Issue by nwconstable** [nwconstable/sdsr-site#15]
**File:** `project/src/train.py`
**Depends on:** model.py (done), build_graph.py (done)

### Context
The centralized baseline trains the WetlandGCN on the full graph. Its loss
curve is the gold standard against which FedAvg and gossip are compared.

### Required function

```python
def train_centralized(
    data: Data,
    model: WetlandGCN,
    epochs: int,
    lr: float = 1e-3,
) -> list[float]:
    """Return per-epoch MSE loss list."""
```

### Acceptance criteria
- [ ] Uses `torch.optim.Adam` and `F.mse_loss`
- [ ] Returns a list of length `epochs`
- [ ] Loss is monotonically decreasing on a small synthetic graph (smoke test)
- [ ] Model state is updated in-place
- [ ] `__main__` block in train.py runs a 10-epoch smoke test with a 10-node graph
